File: api-csharp/Transformations/TransformationBuilder.cs
namespace ImageProcessor.Transformations;

class TransformationBuilder {

    private List<ITransformation> transformations;

    public TransformationBuilder() {
        this.transformations = new List<ITransformation>();
    }

    public void buildTransformations(List<string> transformations) {
        foreach(string transformation in transformations)
            this.transformations.Add(TransformationFactory.getTransformation(transformation));
    }

    public List<ITransformation> getTransformations() {
        return this.transformations;
    }
}

File: api-typescript/src/Transformations/TransformationBuilder.ts
import { ITransformation } from "./ITransformation";
import { TransformationFactory } from "./TransformationFactory";

class TransformationBuilder {

    private transformations: Array<ITransformation>;

    constructor() {
        this.transformations = new Array<ITransformation>();
    }

    public buildTransformations(transformations: Array<string>): void {
        transformations.forEach(t => {
            let transformation = TransformationFactory.getTransformation(t);
            if (transformation)
                this.transformations.push(transformation);
        });
    }

    public getTransformations(): Array<ITransformation> {
        return [...this.transformations];
    }
}

export { TransformationBuilder }